src/test/java/hudson/tasks/JavadocArchiverTest.java

```diff
@@ -73,4 +113,20 @@ private void assertHidden(Action a) {
         }
     }
 
+    private static class CopyResourceDir extends NullSCM {
```

Comment:
I am pretty sure there is already something in `jenkins-test-harness` for this purpose. Easier still is to just use `r.jenkins.getWorkspaceDirFor(p).copyFrom(url)` before starting the build, since by default the existing workspace will be reused.

Reply:
There is, but for decompressing zips only which I refuse to use.